Frontend-unit-tests-guide.md

```diff
@@ -1,409 +1,484 @@
 
 ![Example for branch coverage symbols](https://user-images.githubusercontent.com/34922478/80809691-9340aa80-8b98-11ea-9a22-6fd21171db85.png)
 
-### Rules
-Before starting to write a unit test, you need to be aware of some rules that must be followed in order to make your test consistent and correct and regression-proof.
+#### Ensuring that coverage is maintained
+
+In order to make the coverage stable, by default we require that every frontend file is fully covered. Since we are still increasing frontend coverage, there are some exceptions to this rule. Files that are not fully covered are listed in `scripts/check_frontend_coverage.py`.
 
-Karma is a test runner that is used in the frontend codebase. Karma runs tests by compiling them all at once. This causes a tricky problem: a method in a file might be covered by a test suite for a different file, which can provide a false positive on the coverage report because each file is supposed to be fully covered by its corresponding spec file.
-Thus, when testing a file, make sure that:
-- Each file has its own spec file which tests each line of the file.
-- **You should run the tests locally using fdescribe** in the outer describe of the file, and then make sure that the coverage is 100%. See the [Unit Test Structure](#unit-test-structure) section below.
+If you pass the `--check_coverage` flag when running the tests, then the tests will check whether all expected files are fully covered. The output looks like this:
+
+![Example of output by tracking the changes in the frontend unit tests](https://user-images.githubusercontent.com/34922478/80845732-8182f580-8be0-11ea-9c47-0f46f1f44c2f.png)
+
+## Write frontend tests
 
 ### Unit test structure
+
 A unit test is made of functions that configure the test environment, make assertions, and separate the different contexts of each situation. There are some test functions that are used across the codebase:
-- #### **describe**  
-  The describe function has a string parameter which should contain the name of the component being tested or     (when nested within another describe function) should describe the conditions imposed on the specific context pertaining to the tests in that “describe” block. Here are some examples:
-  ```
+
+#### describe
+
+The describe function has a string parameter which should contain the name of the component being tested or (when nested within another describe function) should describe the conditions imposed on the specific context pertaining to the tests in that “describe” block. Here are some examples:
```

Comment:
Should "describe" be written as `describe` throughout (where its literal English meaning would otherwise not make sense in context)? Similarly for other code keywords.

Reply:
done